lib/src/facts/linux/filesystem_resolver.cc

```diff
@@ -152,7 +155,23 @@ namespace facter { namespace facts { namespace linux {
             }
 
             // Populate the size (the size is given in 512 byte blocks)
-            string blocks = file::read((path("/sys/class/block") / path(part.name).filename() / "/size").string());
+            // Check for /sys/class/block. Fall back to /sys/block if not present.
+            bs::error_code ec;
+            string blocks;
+            auto part_path = path(part.name).filename();
+            if (exists("/sys/class/block", ec) && !ec) {
+              blocks = file::read((path("/sys/class/block") / part_path / "/size").string());
+            } else {
+              boost::regex expression("(\\D+)\\d*");
+              boost::cmatch what;
+              if (boost::regex_match(part_path.string().c_str(), what, expression)) {
```

Comment:
https://github.com/puppetlabs/leatherman/blob/master/util/inc/leatherman/util/regex.hpp#L70 has a re_search helper that might make this a little simpler. You can do things like

```
string device;
if (re_search(path_part.string(), boost::regex("(\\D+)\\d*"), device)) ...
```